src/Sulu/Bundle/ContentBundle/Controller/NodeController.php

```diff
@@ -582,4 +584,13 @@ public function getSecurityContext()
             return 'sulu.webspaces.' . $webspace->getKey();
         }
     }
+
+    /**
+     * Returns the class name of the object to check
+     * @return string
+     */
+    public function getSecuredClass()
+    {
+        return Structure::class;
```

Comment:
I would intuitively expect `$hasPermission = $checker->checkObject($object)` type of API. Without the object maybe `$checker->checkClassId(Structure::class, $id)`.

I might then check the permission to access the object via. EventListeners at the storage layer.